demo/embed_fonts.nim

```diff
@@ -19,35 +19,32 @@ proc createPDF(doc: PDF) =
   let SAMP_TEXT = "The Quick Brown Fox Jump Over The Lazy Dog"
 
   doc.addPage(size, PGO_PORTRAIT)
-  draw_title(doc, "PER-FONT EMBEDDING DEMO")
-
-  # Example 1: Font with individual embedding enabled
-  doc.setFont("Calligrapher", {FS_REGULAR}, 10, ENC_STANDARD, embedFont = true)
-  doc.drawText(15, 30, "Calligrapher - embedded=true")
-
-  # Example 2: Font without individual embedding
-  doc.setFont("Eunjin", {FS_REGULAR}, 10, ENC_STANDARD, embedFont = false)
-  doc.drawText(15, 50, "Eunjin - embedded=false")
-
-  # Example 3: Using convenience method with embedding
-  doc.setFont("KaiTi", 10, embedFont = true)
-  doc.drawText(15, 70, "KaiTi - convenience method")
-
-  # Example 4: Small text with embedding
-  doc.setFont("Calligrapher", {FS_REGULAR}, 8, ENC_STANDARD, embedFont = true)
-  doc.drawText(15, 90, "Sample text with smaller embedded font")
-
-  # Example 5: Base14 font (embedding flag ignored)
-  doc.setFont("Times", {FS_REGULAR}, 10, ENC_STANDARD, embedFont = true)
-  doc.drawText(15, 110, "Times - Base14 ignores flag")
-
-  # Example 6: Show global override behavior
-  doc.setFont("Helvetica", {FS_REGULAR}, 8, ENC_STANDARD, embedFont = false)
-  doc.drawText(15, 130, "Note: Global flag overrides individual settings")
-
-  doc.setInfo(DI_TITLE, "Per-Font Embedding Demo")
+  draw_title(doc, "FONT RENDERING MODES DEMO")
+
+  # Example 1: Font with renderMode=frmEmbed
+  doc.setFillColor(0, 0, 0)  # Set black color for text
+  doc.setFont("Calligrapher", {FS_REGULAR}, 10, ENC_STANDARD, renderMode = frmEmbed)
+  doc.drawText(15, 30, "Calligrapher - renderMode=frmEmbed")
+
+  # Example 2: Korean text with Eunjin font (embedded)
+  doc.setFillColor(0, 0, 0)  # Set black color for text
+  doc.setFont("Eunjin", {FS_REGULAR}, 10, ENC_UTF8, renderMode = frmEmbed)
+  doc.drawText(15, 60, "은진 - 한국어 테스트 (embedded)")
+
+  # Example 3: Path rendering (text drawn as vector paths)
+  doc.setFillColor(0, 0, 0)  # Set black color for text
+  doc.setFont("KaiTi",  {FS_REGULAR}, 10,  enc = ENC_UTF8, renderMode = frmPathRendering)
+  doc.drawText(15, 90, "你好世界 (text drawn as vector paths)")
+
+  # Example 4: Base14 font (embedding flag ignored)
+  doc.setFillColor(0, 0, 0)  # Set black color for text
+  doc.setFont("Times", {FS_REGULAR}, 10, ENC_STANDARD, renderMode = frmEmbed)
+  doc.drawText(15, 120, "Times (Base14 ignores flag)")
+  
+
+  doc.setInfo(DI_TITLE, "Font Rendering Modes Demo")
   doc.setInfo(DI_AUTHOR, "Andri Lim")
-  doc.setInfo(DI_SUBJECT, "Demonstrating individual font embedding")
+  doc.setInfo(DI_SUBJECT, "Demonstrating three font rendering modes: frmDefault, frmEmbed, frmPathRendering")
```

Comment:
You are not including the frmDefault, so only two font's are getting testing. If you need more options for the testing, then create to test files: embed_fonts_default.nim and embed_fonts_options.nim